packages/gatsby/src/utils/map-pages-to-static-query-hashes.ts

```diff
@@ -0,0 +1,130 @@
+import { uniqBy } from "lodash"
+import { IGatsbyState } from "../redux/types"
+
+const mapComponentsToStaticQueryHashes = (
+  staticQueryComponents: IGatsbyState["staticQueryComponents"]
+): Map<string, string> =>
+  Array.from(staticQueryComponents).reduce(
+    (map, [, { componentPath, hash }]) => {
+      map.set(componentPath, hash)
+      return map
+    },
+    new Map()
+  )
+
+export function mapTemplatesToStaticQueryHashes(
+  reduxState: IGatsbyState,
+  compilation
+): Map<string, Array<number>> {
+  const { components, staticQueryComponents } = reduxState
+  const { modules } = compilation
+
+  const terminalNodes = [
+    `.cache/api-runner-browser-plugins.js`,
+    `.cache/async-requires.js`,
+  ]
```

Comment:
Should `.cache/sync-requires.js` be terminal node too? (for `gatsby develop` when we don't make use of `async-requires`)

Reply:
good question, @sidharthachatterjee ?

Reply:
Ah, yup. I need a different set of terminal nodes based on env (and also need to add `api-runner-ssr.js`

Reply:
Added `.cache/sync-requires.js` and `api-runner-ssr.js`